atomate/vasp/firetasks/exchange_tasks.py

```diff
@@ -0,0 +1,458 @@
+            ordering_changed = not np.array_equal(
+                np.sign(input_order_check), np.sign(final_order_check)
+            )
+
+            symmetry_changed = (
+                final_structure.get_space_group_info()[0]
+                != input_structure.get_space_group_info()[0]
+            )
+
+            total_magnetization = abs(
+                d["calcs_reversed"][0]["output"]["outcar"]["total_magnetization"]
+            )
+            num_formula_units = sum(
+                d["calcs_reversed"][0]["composition_reduced"].values()
+            ) / sum(d["calcs_reversed"][0]["composition_unit_cell"].values())
+            total_magnetization_per_formula_unit = (
+                total_magnetization / num_formula_units
+            )
+            total_magnetization_per_unit_volume = (
+                total_magnetization / final_structure.volume
+            )
+
+            summary = {
+                "formula": formula,
+                "formula_pretty": formula_pretty,
+                "parent_structure": self["parent_structure"].as_dict(),
+                "wf_meta": d["wf_meta"],  # book-keeping
+                "task_id": d["task_id"],
+                "structure": final_structure.as_dict(),
+                "magmoms": magmoms,
+                "input": {
+                    "structure": input_structure.as_dict(),
+                    "ordering": input_analyzer.ordering.value,
+                    "symmetry": input_structure.get_space_group_info()[0],
+                    "index": ordering_index,
+                    "origin": ordering_origin,
+                    "input_index": self.get("input_index", None),
+                },
+                "total_magnetization": total_magnetization,
+                "total_magnetization_per_formula_unit": total_magnetization_per_formula_unit,
+                "total_magnetization_per_unit_volume": total_magnetization_per_unit_volume,
+                "ordering": final_analyzer.ordering.value,
+                "ordering_changed": ordering_changed,
+                "symmetry": final_structure.get_space_group_info()[0],
+                "symmetry_changed": symmetry_changed,
+                "energy_per_atom": d["output"]["energy_per_atom"],
+                "stable": stable,
+                "decomposes_to": decomposes_to,
+                "energy_above_ground_state_per_atom": energy_above_ground_state_per_atom,
+                "energy_diff_relax_static": energy_diff_relax_static,
+                "created_at": datetime.utcnow(),
+            }
+
+            if fw_spec.get("tags", None):
+                summary["tags"] = fw_spec["tags"]
+
+            summaries.append(summary)
+
+        mmdb.collection = mmdb.db["magnetic_orderings"]
+        mmdb.collection.insert(summaries)
+
+        logger.info("Magnetic orderings calculation complete.")
+
+
+@explicit_serialize
+class HeisenbergModelMapping(FiretaskBase):
```

Comment:
So this class seems to do two things at once. It computes the the exchange parameters AND adds the model to the database.

If the mapping is trivial then I think it would be best to just treat this is a ToDb FireTask (and rename it accordingly). If the mapping can take a considerable amount of time, say > 10 mins then maybe this could be broken up into two tasks. One to do the mapping and writing the mapping to a file, and other to add the data to the database.

Reply:
The timing for the mapping is highly variable so I'll split it into two tasks.

Reply:
Ok great. Similar to the `BoltztrapToDb` firetask, you can just presume you are in the correct folder with the mapping written to a json file.

Reply:
Done! 👍